packages/bloc/lib/src/bloc_base.dart

```diff
@@ -1,10 +1,46 @@
 part of bloc;
 
+/// An object that provides access to a stream of states over time.
+abstract class Streamable<State> {
+  /// The current [stream] of states.
+  Stream<State> get stream;
+}
+
+/// A [Streamable] that provides synchronous access to the current [state].
+abstract class StateStreamable<State> extends Streamable<State> {
+  /// The current [state].
+  State get state;
+}
+
+/// An object which can emit new states.
+abstract class Emittable<State> {
+  /// Emits a new [state].
+  void emit(State state);
+}
+
+/// An object which must be closed when no longer in use.
+abstract class Closable {
+  /// Closes the current instance.
+  /// The returned future completes when the instance has been closed.
+  Future<void> close();
+}
+
+/// A generic destination for errors.
+///
+/// Multiple errors can be reported to the sink via `addError`.
+abstract class ErrorSink extends Closable {
+  /// Adds an [error] to the sink with an optional [stackTrace].
+  ///
+  /// Must not be called on a closed sink.
+  void addError(Object error, [StackTrace? stackTrace]);
+}
+
 /// {@template bloc_base}
 /// An interface for the core functionality implemented by
 /// both [Bloc] and [Cubit].
 /// {@endtemplate}
-abstract class BlocBase<State> {
+abstract class BlocBase<State>
```

Comment:
This is only style suggestion:
I like type generics like `<State extends Object>` if anything except Null and Void
or `<State extends Object?>` if litteraly anything

Reply:
That will technically be a breaking change at this point because currently you can create a Bloc with a dynamic state. I'm not opposed to making the change in the future though but I don't think it should be done at this point because it is technically breaking.

Reply:
Missundestending)
You can set `BlocBase<State extends Object?>` without breaking changes.
I would do like this. But this is only style point.
If you don’t like it, don’t need to, of course

Reply:
It's technically breaking.

The following is currently allowed but with the change it would not compile:

```dart
class MyBloc extends Bloc<dynamic, dynamic> {...}
```

Reply:
Yeah I can do `Object?` but `Object` would be breaking.